cobalt/h5vcc/h5vcc_metrics.idl

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Defines an interface to support interactions with Cobalt metric logging.
+// For example, a callback API such that the web client can intercept and
+// process metrics sent from Cobalt.
+
+interface H5vccMetrics {
+  // Interface for web client to bind an event handler that will be invoked
+  // every time Cobalt wants to upload a metrics payload. Only one event
+  // handler callback can be registered at any time. Duplicate calls to
+  // onMetricEvent will override any previously registered callback.
+  void onMetricEvent(H5vccMetricEventHandler eventHandler);
+
+  // API to explicitly enable or disable Cobalt metrics logging. Pass true
+  // to turn on and false to disable. If disabled, the metric event handler
+  // should never get called afterward.
+  void toggleMetricsEnabled(boolean isEnabled);
```

Comment:
Maybe it's better to turn this into a read/write boolean attribute .metricsEnabled ? Clients can query and log it as well if needed then.

Again good thing to call out in design doc

Reply:
I kept ToggleMetricsEnabled so I have a hook when it's changed and can enable/disable the metrics service on the fly. I added a "GetMetricsEnabled()" getter for clients to query.

Reply:
A javascript read/write attribute actually also nets you a callable hook - it splits into a getter/setter as generated by the IDL compiler. Separate get/set function calls in JavaScript are quite redundant and less of JS / IDL general style

Reply:
I don't think it's quite this cut and dry, the web api has assignment as well, see i.e. `document.designMode` and `element.contentEditable`. `disableCache`, on the other hand, isn't a standard web api pattern, just one api that we added—so I don't really think it indicates much. Also assignment is more declarative which is in line with React-y patterns, so would probably be more fitting for Kabuki.

Reply:
IDK if pointing to DOM nodes (where attributes are native) is a fair comparison. I also don't know if we should be striving to be like React (I have my own strong opinions about that). I guess Zylon "borrows" from React, but that's a weak relation. 

I've sent an informal poll to Kabuki engineers how they'd like to use these APIs.  At any rate, I don't care _that_ much about this as the end result is the same. I do prefer the function-based approach for the reasons I listed above, but willing to change that if folks feel strongly.

Reply:
Wiz next is almost exactly React, it's a very strong relation. If you had some examples of this pattern elsewhere in the web apis it would help out

Reply:
Yep I agree, just wanted to make sure we weren't deciding on an established pattern, because I agree there isn't one (and that's okay)